src/main/java/Ui.java

```diff
@@ -0,0 +1,148 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a user interface and is in charge of printing contents to screen
+ */
+public class Ui {
+
+    private void welcome(){
+        String logo =   "_|_|_|_|_|    _|_|                _|_|_|      _|_|     _|  \n" +
+                        "    _|      _|    _|              _|    _|  _|    _|   _|  \n" +
+                        "    _|      _|    _|  _|_|_|_|_|  _|    _|  _|    _|   _|  \n" +
+                        "    _|      _|    _|              _|    _|  _|    _|       \n" +
+                        "    _|        _|_|                _|_|_|      _|_|     _|  ";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("This is a Duke Project.\n");
+        System.out.println("What you are going to do today?");
+    }
+
+    /**Reads incoming commands and use a parser to pase it.
+     *
+     * @param parser a parser used to parse incoming commands
+     * @return the User interface itself
+     */
+    public Ui read(Parser parser){
+        welcome();
+        Scanner sc = new Scanner(System.in);
+        Command command = Command.INIT;
+        while (command != Command.BYE){      //when last command was bye
+            System.out.print("> ");
+            command = Command.getCommand(sc.nextLine());
+            try {
+                parser.runCommand(command);
+            } catch (Exception e){
+                showExceptionContent(e.getMessage());
+            }
+        }
+        sc.close();
+        return this;
+    }
+
+    /**
+     * Prints all the tasks stored in memory
+     *
+     * @param todoList a list contains all the tasks
+     */
+    static public void printList(ArrayList<Task> todoList){
+        for (int i = 0; i < todoList.size(); i++){
+            Task task = todoList.get(i);
+            if (task.checkDone())
+                print(i+1 + ".["+task.getTaskType()+"][X] " + task.toString());
+            else
+                print(i+1 + ".["+task.getTaskType()+"][ ] " + task.toString());
+        }
+    }
+
+    static public void print(String str){
+        for (String s:str.split("\n"))
+            System.out.println("    " + s);
+    }
+
+    public void showExceptionContent(String err){
+        print(err);
+    }

```

Comment:
Can I check why this method is needed?

Can't the static print method just be called directly?